test/Dialect/BaseModelica/Transforms/CallCSE/basic-cse.mlir

```diff
@@ -0,0 +1,61 @@
+// RUN: modelica-opt %s --call-cse | FileCheck %s
+
+// CHECK-LABEL: @Test
+module @Test {
+	bmodelica.function @foo {
+		bmodelica.variable @x : !bmodelica.variable<f64, input>
+		bmodelica.variable @y : !bmodelica.variable<f64, output>
+
+		bmodelica.algorithm {
+			%0 = bmodelica.variable_get @x : f64
+			bmodelica.variable_set @y, %0 : f64
+		}
+	}
+
+    // CHECK: bmodelica.model
+	bmodelica.model @M {
+		// CHECK: bmodelica.variable @[[CSE:_cse0_0]]
+		bmodelica.variable @x : !bmodelica.variable<f64>
+		bmodelica.variable @y : !bmodelica.variable<f64>
+
+		%t0 = bmodelica.equation_template inductions = [] {
+			%0 = bmodelica.variable_get @x : f64
+			%lhs = bmodelica.equation_side %0 : tuple<f64>
+			%1 = bmodelica.constant 1.0 : f64
+			// CHECK: %[[RES0:.*]] = bmodelica.variable_get @[[CSE]]
+			// CHECK-NEXT: %[[LHS0:.*]] = bmodelica.equation_side %[[RES0]]
+			// CHECK-NEXT: bmodelica.equation_sides %{{.*}}, %[[LHS0]]
+			%2 = bmodelica.call @foo(%1) : (f64) -> f64
+			%rhs = bmodelica.equation_side %2 : tuple<f64>
+			bmodelica.equation_sides %lhs, %rhs : tuple<f64>, tuple<f64>
+		}
+
+		%t1 = bmodelica.equation_template inductions = [] {
+			%0 = bmodelica.variable_get @y : f64
+			%lhs = bmodelica.equation_side %0 : tuple<f64>
+			%1 = bmodelica.constant 1.0 : f64
+			// CHECK: %[[RES1:.*]] = bmodelica.variable_get @[[CSE]]
+			// CHECK-NEXT: %[[LHS1:.*]] = bmodelica.equation_side %[[RES1]]
+			// CHECK-NEXT: bmodelica.equation_sides %{{.*}}, %[[LHS1]]
+			%2 = bmodelica.call @foo(%1) : (f64) -> f64
+			%rhs = bmodelica.equation_side %2 : tuple<f64>
+			bmodelica.equation_sides %lhs, %rhs : tuple<f64>, tuple<f64>
+		}
+
+		bmodelica.dynamic {
+			bmodelica.equation_instance %t0 : !bmodelica.equation
+			bmodelica.equation_instance %t1 : !bmodelica.equation
+		}
+
+		// CHECK: %[[TEMPLATE:.*]] = bmodelica.equation_template
```

Comment:
The inductions property is missing, as well as the opening (and later closing) of the region. Also indent the expected IR.

Reply:
The reason I didn't include those kinds of things is due to what I read here: https://mlir.llvm.org/getting_started/TestingGuide/#filecheck-best-practices 

I'll add them if you prefer though

Reply:
Makes sense, keep it the way you did for what regards the region.
I would still add the inductions property though, that's something you decide on while creating the operation within the pass, and for this reason worth to be checked.